src/sentry/utils/circuit_breaker.py

```diff
@@ -0,0 +1,44 @@
+from datetime import datetime
+
+from django.core.cache import cache
+
+from sentry.utils import metrics
+
+DEFAULT_ERROR_LIMIT = 30
```

Comment:
Do we want to be smarter and look at an overall (hourly?) error rate instead of consecutive errors? As it is, this only works for complete failures, so if the occasional request succeeds the breaker wouldn't be activated.

Reply:
@snigdhas I think consecutive errors is fine because the primary use case we are solving for is seer going down.